akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -352,35 +358,51 @@ private[akka] class Shard(
           // and this could always fail before ddata store completes retrying writes
           settings.tuningParameters.updatingStateTimeout)
 
-        context.become {
-          case RememberEntitiesShardStore.UpdateDone(`entityId`) =>
-            if (VerboseDebug) log.debug("Update of [{}] {} done", entityId, command)
+        context.become(waitingForUpdate(Map.empty))
+
+        def waitingForUpdate(pendingStarts: Map[EntityId, Option[ActorRef]]): Receive = {
+          // none of the current impls will send back a partial update, yet!
+          case RememberEntitiesShardStore.UpdateDone(ids) if ids == ids =>
```

Comment:
Perhaps the comparison is not worth it anymore when there is a set to compare with another set. I think I mostly put it there to detect bugs and it was cheap (if we really need to detect that we accidentally triggered two writes concurrently we could do that some other way?).

Reply:
Okay i'll remove